src/App.js

```diff
@@ -0,0 +1,33 @@
+import React, { Component } from 'react';
+import './App.css';
+
+class App extends Component {
+  state = {
+    tasks: ['something'],
+    value: '',
+  }
+
+  addTask = () => this.setState(() => !!this.state.value.trimRight() && !this.state.tasks.includes(this.state.value) ? this.state.tasks.push(this.state.value) : null)
```

Comment:
you shouldn't rely on `this.state` inside of the `this.setState`, due to the fact, that this.setState has async nature. 
function passed as an argument to the `this.setState` gets the props `prevState, prevProps`. 
so it should be refactored to: 
`this.setState(({ prevState, prevProps }) => !!prevState.value.trim() &&...)`
you can also do in ES6 + way instead of old hacking and do `Boolean(prevState.value.trim())`.